contracts/apps/TokenIOU.sol

```diff
@@ -0,0 +1,214 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+
+import "@openzeppelin/contracts/interfaces/IERC1271.sol";
+
+contract TokenIOU is Initializable, OwnableUpgradeable, UUPSUpgradeable {
+    IERC20Upgradeable public token;
+
+    /// only a single redeem per hash
+    ///
+    /// allows us to check which username was already reserved by a profile
+    mapping(bytes32 redeemHash => uint48 time) public redeemed;
+
+    // Redeem event
+    event Redeem(address indexed from, address indexed to, uint256 amount);
+
+    function initialize(IERC20Upgradeable _token) public initializer {
+        __Ownable_init();
+        __UUPSUpgradeable_init();
+        token = _token;
+    }
+
+    /**
+        * @notice Get the hash of the redeem request
+        * @param from       The address of the redeemer
```

Comment:
Isn't this supposed to be the issuer?

(also redeemer feels too catholic) ;-)

Reply:
Copilot generated docs :P